src/shared/loop-util.c

```diff
@@ -220,6 +223,21 @@ static int loop_configure_fallback(int fd, const struct loop_config *c) {
                 if (ioctl(fd, BLKFLSBUF, 0) < 0)
                         log_debug_errno(errno, "Failed to issue BLKFLSBUF ioctl, ignoring: %m");
 
+        /* If a block size is requested then try to configure it. If that doesn't work, ignore errors, but
+         * afterwards, let's validate what is in effect, and if it doesn't match what we want, fail */
+        if (c->block_size != 0) {
+                uint32_t ssz;
+
+                if (ioctl(fd, LOOP_SET_BLOCK_SIZE, (unsigned long) c->block_size) < 0)
+                        log_debug_errno(errno, "Failed to set sector size, ignoring: %m");
+
+                r = blockdev_get_sector_size(fd, &ssz);
```

Comment:
did you forget an 'else' here? or a return above?

Reply:
nope. if the ioctl fails, then that's noteworthy so that we debug log it. but, we immediately check what the sector size is that is in effect. If this is the right size anyway, then we can just ignore the previous error. Only if it is not the right size anyway we have to fail.